### PR TITLE
[Localization] Localizable owned money symbol on battle UI

### PR DESCRIPTION
**Pull request by Adri1**
## What are the changes the user will see?
The money displayed during battle has now its money symbol localizable

## Why am I making these changes?
To allow langauges to move the money symbol (or to change it if needed)

## What are the changes from a developer perspective?
- battle-scene.ts in src, edited to make the money owned localizable
- Addition of battle-scene.ts translation file in all locales
- Added battle-scene.ts compatibility to all locales config.ts 

### Screenshots/Videos
**BEFORE (French) :**
image

**AFTER (French) :**
image

## How to test the changes?
Edit the text in a local battle-scene.ts and play the game in that locale

## Checklist
- [X] **I'm using `beta` as my base branch**
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [X] If I have text, did I add placeholders for them in locales?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/de/battle-scene.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const battleScene: SimpleTranslationEntries = {
+  "moneyOwned": "₽{{formattedMoney}}"
```
Comment (CodeTappert) [inline src/locales/de/battle-scene.ts:4]:
```suggestion
  "moneyOwned": "{{formattedMoney}} ₽"

```

**Comment by frutescens** [comment]
Could you make sure the localization files match up to the formats described here? https://bulbapedia.bulbagarden.net/wiki/Pok%C3%A9mon_Dollar#Formatting 

I know there are some languages missing, but just the ones that are available.

**Comment by Adri1** [comment]
> Could you make sure the localization files match up to the formats described here? https://bulbapedia.bulbagarden.net/wiki/Pok%C3%A9mon_Dollar#Formatting
> 
> I know there are some languages missing, but just the ones that are available.

I already sublitted it to my other translators collegues anyways but thank for it, didn't knew bulba made a inventory about formatting

**Review by frutescens — APPROVED** [review]

**Review by NicusPulcis — APPROVED** [review]
italian approved

**Review by EnochG1 — APPROVED** [review]

File: src/locales/ko/battle-scene.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const battleScene: SimpleTranslationEntries = {
+  "moneyOwned": "{{formattedMoney}}원"
```
Comment (EnochG1) [inline src/locales/ko/battle-scene.ts:4]:
```suggestion
  "moneyOwned": "₽{{formattedMoney}}"
```
Korean currency is '원'
But we are using ₽ at other locale texts

https://github.com/pagefaultgames/pokerogue/blob/main/src/locales/ko/modifier-select-ui-handler.ts#L12

https://github.com/pagefaultgames/pokerogue/blob/main/src/locales/ko/battle.ts#L16

**Comment by frutescens** [comment]
@EnochG1 My Korean is not too great so I want to double-check my changes with you. 
` "moneyWon": "상금으로\n{{moneyAmount}}원을 손에 넣었다!",
  "moneyPickedUp": "{{moneyAmount}}원을 주웠다!",`

Would it be {{moneyAmount}}원을 주웠다! or {{moneyAmount}}원 을 주웠다! ?

**Comment by EnochG1** [comment]
> @EnochG1 My Korean is not too great so I want to double-check my changes with you. ` "moneyWon": "상금으로\n{{moneyAmount}}원을 손에 넣었다!", "moneyPickedUp": "{{moneyAmount}}원을 주웠다!",`
> 
> Would it be {{moneyAmount}}원을 주웠다! or {{moneyAmount}}원 을 주웠다! ?

`{{moneyAmount}}원을 주웠다!` is correct.

In fact, two sentences above were examples in battle.ts,
and we're now using several sentences with ₽.

```Typescript
  "rerollCost": "₽{{formattedMoney}}",
  "itemCost": "₽{{formattedMoney}}"

  "MoneyAchv": {
    description: "누적 소지금 ₽{{moneyAmount}} 달성",
  },

  "moneyWon": "상금으로\n₽{{moneyAmount}}을 손에 넣었다!",
  "moneyPickedUp": "₽{{moneyAmount}}을 주웠다!",

    "MoneyRewardModifierType": {
      description: "{{moneyMultiplier}} 양의 돈을 획득한다. (₽{{moneyAmount}})",
      extra: {
        "small": "적은",
        "moderate": "적당한",
        "large": "많은",
      },
    },

  "moneyInterestApply": "{{typeName}}[[로]]부터\n₽{{moneyAmount}}[[를]] 받았다!",
```

Now discussing with the Korean translation team whether to use `₽` or `원`, 
I think it should be the same in Korean either way.

**Comment by EnochG1** [comment]
@Adri1 @frutescens 
We've talked about preferring `원` as it was in the original game,
but changing 8 sentences in this PR doesn't seem to fit the checklist (can't be split into smaller PRs),
so we're going to work on it separately.
Thank you for your help :)

**Comment by frutescens** [comment]
That sounds good!

**Review by frutescens — APPROVED** [review]
Tested + Approved
Italian / German / Spanish (same format)
image

Japanese
image

**Review by KimJeongSun — APPROVED** [review]
Regarding the Korean translation, I agree with the existing Korean translator @EnochG1 's opinion. 
The code update itself seems necessary, so it looks good.

**Review by DayKev — APPROVED** [review]
EN:
image
PT_BR:
image
KO:
image
ZH_CN:
image
ZH_TW:
image

**Pull request merged**